src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -202,27 +202,30 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     void TermControlAutomationPeer::NotifyNewOutput(std::wstring_view newOutput)
     {
-        // Try to suppress any events (or event data)
-        // that is just the keypress the user made
         auto sanitized{ Sanitize(newOutput) };
-        while (!_keyEvents.empty() && IsReadable(sanitized))
+        // Try to suppress any events (or event data)
+        // that are just the keypresses the user made
         {
-            if (til::toupper_ascii(sanitized.front()) == _keyEvents.front())
-            {
-                // the key event's character (i.e. the "A" key) matches
-                // the output character (i.e. "a" or "A" text).
-                // We can assume that the output character resulted from
-                // the pressed key, so we can ignore it.
-                sanitized = sanitized.substr(1);
-                _keyEvents.pop_front();
-            }
-            else
+            auto keyEvents = _keyEvents.lock();

```

Comment:
you might need to make this `auto&` or even (C++14) `decltype(auto)`. I _think_ that `auto keyEvents` will result in a _copy_ of `keyEvents` (thanks to how auto works).

<img width="207" alt="image" src="https://user-images.githubusercontent.com/189190/213035815-2435ffac-8862-48fb-90d6-d8e3a4156ac3.png">

<img width="206" alt="image" src="https://user-images.githubusercontent.com/189190/213035825-0830ad79-ec92-42df-a67b-67f2b64027f3.png">

See how the first one compiles down to `escape(9)` even though we just set the value to `7` through `auto m`!

Reply:
More info at https://blog.petrzemek.net/2017/12/08/when-auto-seemingly-deduces-a-reference-in-cpp/

Reply:
Oh, but this returns a proxy object... so really it doesn't matter. You're actually using a `guard<deque>`, which is fine. Ignore me! You have not been bitten by C++; in fact, I have!